luaui/Widgets/api_object_spotlight.lua

```diff
@@ -0,0 +1,549 @@
+			0, 0, 0, 0, -- { id = 5, name = "instData", size = 4, type = GL.UNSIGNED_INT }
+			instanceWorldPosOverride[1], instanceWorldPosOverride[2], instanceWorldPosOverride[3], -- { id = 8, name = "worldPosOverride", size = 3 },
+		},
+		objectInstanceIDs[type][objectID][owner],
+		true,
+		false,
+		instanceObjectID
+	)
+end
+
+local function removeSpotlight(type, owner, objectID)
+	if not spotlightTypes[type] then
+		error("invalid spotlight target type: " .. (type or "<nil>"))
+	end
+
+	if not objectInstanceIDs[type][objectID] or not objectInstanceIDs[type][objectID][owner] then
+		return
+	end
+
+	-- owner
+	objectOwners[type][objectID][owner] = nil
+
+	-- instance
+	if instanceVBOs[type].instanceIDtoIndex[objectInstanceIDs[type][objectID][owner]] then
+		popElementInstance(instanceVBOs[type], objectInstanceIDs[type][objectID][owner], false)
+	end
+	objectInstanceIDs[type][objectID][owner] = nil
+
+	-- duration
+	if objectExpireTimes[type][objectID] and objectExpireTimes[type][objectID][owner] then
+		objectExpireTimes[type][objectID][owner] = nil
+	end
+end
+
+function widget:UnitDestroyed(unitID, unitDefID, unitTeam, attackerID, attackerDefID, attackerTeam)
+	if objectOwners["unit"][unitID] then
+		for owner in pairs(objectOwners["unit"][unitID]) do
+			removeSpotlight("unit", owner, unitID)
+		end
+	end
+end
+
+function widget:FeatureDestroyed(featureID, allyTeamID)
+	if objectOwners["feature"][featureID] then
+		for owner in pairs(objectOwners["feature"][featureID]) do
+			removeSpotlight("feature", owner, featureID)
+		end
+	end
+end
+
+local t = 0
+function widget:Update(dt)
+	t = t + dt
+	if t < REMOVE_EXPIRED_SPOTLIGHTS_PERIOD then
+		return
+	end
+	t = 0
+
+	-- remove expired spotlights
+	local toRemove = {}
+	local gs = Spring.GetDrawSeconds()
+	for type, objectOwnerTimes in pairs(objectExpireTimes) do
+		for objectID, ownerTimes in pairs(objectOwnerTimes) do
+			for owner, expireTime in pairs(ownerTimes) do
+				if gs > expireTime then
+					table.insert(toRemove, { type, owner, objectID })
+				end
+			end
+		end
+	end
+
+	for _, removeParams in ipairs(toRemove) do
+		removeSpotlight(unpack(removeParams))
+	end
+end
+
+function widget:DrawWorld()
+	if Spring.IsGUIHidden() then
+		return
+	end
+
+	gl.DepthTest(GL.LEQUAL)
+
+	gl.Blending(GL.SRC_ALPHA, GL.ONE_MINUS_SRC_ALPHA)
+
+	shader:Activate()
+
+	for type, vbo in pairs(instanceVBOs) do
+		if vbo.usedElements > 0 then
+			vbo.VAO:DrawArrays(
+				GL.TRIANGLE_STRIP,
+				vbo.numVertices,
+				0,
+				vbo.usedElements,
+				0
+			)
+		end
+	end
+
+	shader:Deactivate()
+
+	-- other widgets are lazy, so let's reset it to a safe default
+	gl.Blending(GL.SRC_ALPHA, GL.ONE_MINUS_SRC_ALPHA)
+end
+
+function widget:Initialize()
+	if not gl.CreateShader then
+		-- no shader support, so just remove the widget itself, especially for headless
+		widgetHandler:RemoveWidget()
+		return
+	end
+
+	if not initGL4() then
+		widgetHandler:RemoveWidget()
+		return
+	end
+
+	WG["object_spotlight"] = {
```

Comment:
```suggestion
	WG["ObjectSpotlight"] = {
```
Code style convention is PascalCase for WG and GG entries.

Reply:
```suggestion
	WG["viceCityStyleMarker"] = {
```